.github/workflows/release-branch.yaml

```diff
@@ -1,75 +1,110 @@
-          git fetch origin "$tag"
-          git checkout "$tag"
-        else 
-          if [ "${{ github.ref_type }}" != "tag" ]; then
-            >&2 echo "please run workflow on desired tag to create a release branch for or specify a tag as input"
+          
+          
+          branch="releases/$minor"
+          
+          if git ls-remote --exit-code origin refs/heads/$branch ; then
+            >&2 echo "branch $branch already exists, aborting"
             exit 1
           fi
-        
-          tag="${{ github.ref_name }}"
-        fi 
-        
-        if ! [[ "$tag" =~ ^v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-          >&2 echo "no valid non-pre-release tag $tag"
-          exit 1
-        fi
-        if [ "$tag" == "${tag%.0}" ]; then
-          >&2 echo "please use a non-patch tag"
-          exit 1
-        fi
-        if git ls-remote --exit-code origin refs/heads/releases/$tag ; then
-          >&2 echo "branch releases/$tag already exists"
-          exit 1
-        fi
-        echo "creating release branch for $tag"
-        n="releases/$tag"
-        git checkout -b "$n"
-        v="$(go run ./api/version/generate bump-patch)"
-        echo "$v" > VERSION
-        git add VERSION
-        git commit -m "Prepare Development of v$v"
-        git push origin "$n"
+          
+          echo "creating release branch: $branch"
+          
+          git checkout -b "$branch"
+          git push origin $branch
+  bump-main-pr:
+    name: Bump Main Branch to the next Minor Version
+    runs-on: ubuntu-latest
+    needs: create-branch # wait for the release branch to be created, then create a version bump
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.OCMBOT_APP_ID }}
+          private_key: ${{ secrets.OCMBOT_PRIV_KEY }}
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+          token: ${{ steps.generate_token.outputs.token }}
+      - name: Create Bump PR
+        run: |
+          set -e
+          git config --global user.name github-actions
+          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
+          
+          echo "determining next version"
+          version=$(go run ./api/version/generate bump-version)
+          
+          branch="chore/bump-main/$version"
+          
+          git checkout -b chore/bump-main-to-$branch
+          
+          echo "bumping main branch to $version"
+          echo $version > VERSION
+
+          # Trigger a bump of any potential files that depend on a new version
+          make -f hack/Makefile mdref && make -f hack/Makefile go-bindata && make generate
+          
+          git commit -m "Bump main branch to $version after branch cutoff"
+          
+          gh pr create \
+            --fill-first \
+            --base main \
+            --head $branch
```

Comment:
Wouldn't it make sense to reuse the [peter-evans/create-pull-request@v7](https://github.com/open-component-model/ocm/blob/main/.github/workflows/flake_vendorhash.yaml#L52) GHA? It's able now to also create `sign-commits: true`

Reply:
Let me try it out and see. I dont see why we cant use it :) thanks for the pointer